partner_multi_company/models/res_partner.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# © 2015 Oihane Crucelaegui
+# © 2015-2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html.html
+
+from openerp import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _default_company_ids(self):
+        company_model = self.env['res.company']
+        return [(6, 0, company_model._company_default_get('res.partner').ids)]
+
+    company_ids = fields.Many2many(
+        comodel_name='res.company', string="Companies",
+        default=_default_company_ids)
+    company_id = fields.Many2one(
+        comodel_name='res.company', compute="_compute_company_id", store=True,
+        inverse="_inverse_company_id")
+
+    def _amend_company_id(self, vals):
+        if 'company_ids' in vals:
+            if not vals['company_ids']:
+                vals['company_id'] = False
+            else:
+                for item in vals['company_ids']:
+                    if item[0] in (1, 4):
+                        vals['company_id'] = item[1]
+                    elif item[0] in (2, 3, 5):
```

Comment:
Supposedly this is covered by the line https://github.com/OCA/multi-company/pull/33/files#diff-529f207f5b8f18283bc6ced813ddf648R64, but it seems to be converted. Let me check, and if not, I masked it that line.